dhcpv4/bsdp/boot_image.go

```diff
@@ -79,11 +79,10 @@ type BootImage struct {
 }
 
 // ToBytes converts a BootImage to a slice of bytes.
-func (b *BootImage) ToBytes() []byte {
-	bytes := b.ID.ToBytes()
-	bytes = append(bytes, byte(len(b.Name)))
-	bytes = append(bytes, []byte(b.Name)...)
-	return bytes
+func (b BootImage) Marshal(buf *uio.Lexer) {
```

Comment:
Why not using a reference of `BootImage`?

Reply:
b doesn't need to be modified - i think of this like a const in c++. compiler can still optimize it with a reference.

Reply:
I would use a reference even if it doesn't need to be modified but I don't have a strong opinion about it.